src/core_modules/capture-core-utils/rulesEngine/rulesEngine.types.js

```diff
@@ -252,34 +267,39 @@ export type D2FunctionParameters = {
     dhisFunction: any
 }
 export type D2Functions = {
+    'd2:ceil': D2FunctionParameters,
+    'd2:floor': D2FunctionParameters,
+    'd2:round': D2FunctionParameters,
+    'd2:modulus': D2FunctionParameters,
+    'd2:zing': D2FunctionParameters,
+    'd2:oizp': D2FunctionParameters,
     'd2:concatenate': D2FunctionParameters,
     'd2:daysBetween': D2FunctionParameters,
     'd2:weeksBetween': D2FunctionParameters,
     'd2:monthsBetween': D2FunctionParameters,
     'd2:yearsBetween': D2FunctionParameters,
-    'd2:floor': D2FunctionParameters,
-    'd2:modulus': D2FunctionParameters,
     'd2:addDays': D2FunctionParameters,
-    'd2:zing': D2FunctionParameters,
-    'd2:oizp': D2FunctionParameters,
     'd2:count': D2FunctionParameters,
-    'd2:countIfZeroPos': D2FunctionParameters,
     'd2:countIfValue': D2FunctionParameters,
-    'd2:ceil': D2FunctionParameters,
-    'd2:round': D2FunctionParameters,
+    'd2:countIfZeroPos': D2FunctionParameters,
     'd2:hasValue': D2FunctionParameters,
-    'd2:lastEventDate': D2FunctionParameters,
+    // d2:zpvc(<object>, <object>, ...)
     'd2:validatePattern': D2FunctionParameters,
-    'd2:addControlDigits': D2FunctionParameters,
-    'd2:checkControlDigits': D2FunctionParameters,
-    'd2:round': D2FunctionParameters,
-    'd2:lastEventDate': D2FunctionParameters,
     'd2:left': D2FunctionParameters,
     'd2:right': D2FunctionParameters,
     'd2:substring': D2FunctionParameters,
     'd2:split': D2FunctionParameters,
-    'd2:zScoreWFA': D2FunctionParameters,
     'd2:length': D2FunctionParameters,
+    // d2:inOrgUnitGroup(<orgunit_group_code>)
```

Comment:
Why did you comment these D2 functions ? Thanks!

Reply:
I compared the list of D2-functions that can be selected in a program rule expression with the corresponding list in the rules engine, and found that the rules engine was lacking a few. I inserted the missing functions as comments for later reference.